pkg/controllers/cassandra/role/control.go

```diff
@@ -44,8 +44,15 @@ func (c *control) Sync(cluster *v1alpha1.CassandraCluster) error {
 	_, err := c.roleLister.
 		Roles(newRole.Namespace).
 		Get(newRole.Name)
-	if k8sErrors.IsNotFound(err) {
-		_, err = client.Create(newRole)
+	if err == nil {
+		return nil
+	}
+	if !k8sErrors.IsNotFound(err) {
+		return err
+	}
+	_, err = client.Create(newRole)
+	if k8sErrors.IsAlreadyExists(err) {
+		return nil
```

Comment:
We should probably return an error here still, as the resource may not/probably isn't like `newRole` (because it already exists). Then on the next sync, the `IsNotFound` check will be run again and the appropriate action taken.